diff -aurp poppler-0.72.0-orig/CMakeLists.txt poppler-0.72.0/CMakeLists.txt
--- poppler-0.72.0-orig/CMakeLists.txt	2018-12-10 10:32:37.091040000 +0000
+++ poppler-0.72.0/CMakeLists.txt	2018-12-10 10:36:19.530546300 +0000
@@ -37,7 +37,7 @@ endif()
 
 set(POPPLER_MAJOR_VERSION "0")
 set(POPPLER_MINOR_VERSION "72")
-set(POPPLER_MICRO_VERSION "0")
+set(POPPLER_MICRO_VERSION "1")
 set(POPPLER_VERSION "${POPPLER_MAJOR_VERSION}.${POPPLER_MINOR_VERSION}.${POPPLER_MICRO_VERSION}")
 
 set (CMAKE_CXX_STANDARD 14)
diff -aurp poppler-0.72.0-orig/cpp/poppler-global.cpp poppler-0.72.0/cpp/poppler-global.cpp
--- poppler-0.72.0-orig/cpp/poppler-global.cpp	2018-12-10 10:32:37.109040000 +0000
+++ poppler-0.72.0/cpp/poppler-global.cpp	2018-12-10 10:32:58.368040000 +0000
@@ -226,7 +226,11 @@ byte_array ustring::to_utf8() const
         return byte_array();
     }
 
-    MiniIconv ic("UTF-8", "UTF-16");
+#ifdef WORDS_BIGENDIAN
+    MiniIconv ic("UTF-8", "UTF-16BE");
+#else
+    MiniIconv ic("UTF-8", "UTF-16LE");
+#endif
     if (!ic.is_valid()) {
         return byte_array();
     }
@@ -274,7 +278,11 @@ ustring ustring::from_utf8(const char *s
         }
     }
 
-    MiniIconv ic("UTF-16", "UTF-8");
+#ifdef WORDS_BIGENDIAN
+    MiniIconv ic("UTF-16BE", "UTF-8");
+#else
+    MiniIconv ic("UTF-16LE", "UTF-8");
+#endif
     if (!ic.is_valid()) {
         return ustring();
     }
diff -aurp poppler-0.72.0-orig/cpp/poppler-page.cpp poppler-0.72.0/cpp/poppler-page.cpp
--- poppler-0.72.0-orig/cpp/poppler-page.cpp	2018-12-10 10:32:37.106040000 +0000
+++ poppler-0.72.0/cpp/poppler-page.cpp	2018-12-10 10:33:12.163810000 +0000
@@ -365,7 +365,7 @@ std::vector<text_box> page::text_list()
             TextWord *word = word_list->get(i);
 
             std::unique_ptr<GooString> gooWord{word->getText()};
-            ustring ustr = detail::unicode_GooString_to_ustring(gooWord.get());
+            ustring ustr = ustring::from_utf8(gooWord->c_str());
 
             double xMin, yMin, xMax, yMax;
             word->getBBox(&xMin, &yMin, &xMax, &yMax);
diff -aurp poppler-0.72.0-orig/cpp/poppler-private.cpp poppler-0.72.0/cpp/poppler-private.cpp
--- poppler-0.72.0-orig/cpp/poppler-private.cpp	2018-12-10 10:32:37.139040000 +0000
+++ poppler-0.72.0/cpp/poppler-private.cpp	2018-12-10 10:32:58.370040000 +0000
@@ -93,7 +93,7 @@ ustring detail::unicode_GooString_to_ust
 
 ustring detail::unicode_to_ustring(const Unicode *u, int length)
 {
-    ustring str(length * 2, 0);
+    ustring str(length, 0);
     ustring::iterator it = str.begin();
     const Unicode *uu = u;
     for (int i = 0; i < length; ++i) {
