diff -aurp poppler-0.72.0-orig/CMakeLists.txt poppler-0.72.0/CMakeLists.txt
--- poppler-0.72.0-orig/CMakeLists.txt	2018-12-09 14:40:02.261109200 +0000
+++ poppler-0.72.0/CMakeLists.txt	2018-12-09 14:44:48.176108500 +0000
@@ -485,7 +485,7 @@ else()
 add_library(poppler ${poppler_SRCS})
 endif()
 set_target_properties(poppler PROPERTIES VERSION 83.0.0 SOVERSION 83)
-if(MINGW)
+if(MINGW AND BUILD_SHARED_LIBS)
     get_target_property(POPPLER_SOVERSION poppler SOVERSION)
     set_target_properties(poppler PROPERTIES SUFFIX "-${POPPLER_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif()
diff -aurp poppler-0.72.0-orig/cpp/CMakeLists.txt poppler-0.72.0/cpp/CMakeLists.txt
--- poppler-0.72.0-orig/cpp/CMakeLists.txt	2018-12-09 14:40:01.596109200 +0000
+++ poppler-0.72.0/cpp/CMakeLists.txt	2018-12-09 14:45:06.631108500 +0000
@@ -25,7 +25,7 @@ set(poppler_cpp_SRCS
 
 add_library(poppler-cpp ${poppler_cpp_SRCS})
 set_target_properties(poppler-cpp PROPERTIES VERSION 0.5.0 SOVERSION 0)
-if(MINGW)
+if(MINGW AND BUILD_SHARED_LIBS)
     get_target_property(POPPLER_CPP_SOVERSION poppler-cpp SOVERSION)
     set_target_properties(poppler-cpp PROPERTIES SUFFIX "-${POPPLER_CPP_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif()
diff -aurp poppler-0.72.0-orig/glib/CMakeLists.txt poppler-0.72.0/glib/CMakeLists.txt
--- poppler-0.72.0-orig/glib/CMakeLists.txt	2018-12-09 14:40:02.016109200 +0000
+++ poppler-0.72.0/glib/CMakeLists.txt	2018-12-09 14:45:26.289108500 +0000
@@ -90,7 +90,7 @@ set_target_properties(poppler-glib PROPE
 set_target_properties(poppler-glib PROPERTIES C_VISIBILITY_PRESET hidden)
 set_target_properties(poppler-glib PROPERTIES CXX_VISIBILITY_PRESET hidden)
 set_target_properties(poppler-glib PROPERTIES VISIBILITY_INLINES_HIDDEN 1)
-if(MINGW)
+if(MINGW AND BUILD_SHARED_LIBS)
     get_target_property(POPPLER_GLIB_SOVERSION poppler-glib SOVERSION)
     set_target_properties(poppler-glib PROPERTIES SUFFIX "-${POPPLER_GLIB_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif()
