diff --git a/cmake/FindPkgLibraries.cmake b/cmake/FindPkgLibraries.cmake
index 220bb2ce2..39f74a8bc 100644
--- a/cmake/FindPkgLibraries.cmake
+++ b/cmake/FindPkgLibraries.cmake
@@ -14,13 +14,13 @@ function(FIND_PKGLIBRARIES prefix package)
 	foreach(LIBRARY ${${prefix}_LIBRARIES})
 		find_library(${LIBRARY}_RESOLVED ${LIBRARY} PATHS ${${prefix}_LIBRARY_DIRS})
 		if(${${LIBRARY}_RESOLVED} STREQUAL "${LIBRARY}_RESOLVED-NOTFOUND")
-			message(FATAL_ERROR "could not resolve ${LIBRARY}")
+			message(WARNING "could not resolve ${LIBRARY}")
 		endif()
 		list(APPEND RESOLVED_LIBRARIES ${${LIBRARY}_RESOLVED})
 	endforeach()
 
 	set(${prefix}_FOUND 1 PARENT_SCOPE)
-	set(${prefix}_LIBRARIES ${RESOLVED_LIBRARIES} PARENT_SCOPE)
+	set(${prefix}_LIBRARIES ${${prefix}_LIBRARIES} PARENT_SCOPE)
 	set(${prefix}_INCLUDE_DIRS ${${prefix}_INCLUDE_DIRS} PARENT_SCOPE)
 	set(${prefix}_LDFLAGS ${${prefix}_LDFLAGS} PARENT_SCOPE)
 
