diff --git a/configure.ac b.configure.ac
index 33ce2bc..g242ab1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -185,7 +185,7 @@
     AC_CHECK_LIB(seccomp, seccomp_init)
 fi
 if test "$MINGW" = 1; then
-  AC_CHECK_LIB(gnurx,regexec,,AC_MSG_ERROR([libgnurx is required to build file(1) with MinGW]))
+  AC_CHECK_LIB(gnurx,regexec,,LIBS="$LIBS -ltre -lintl -liconv")
 fi
 
 dnl See if we are cross-compiling
