diff -aurp librsvg-2.48.8-orig/configure.ac librsvg-2.48.8/configure.ac
--- librsvg-2.48.8-orig/configure.ac	2020-10-09 16:23:53.315550200 +0000
+++ librsvg-2.48.8/configure.ac	2020-10-09 17:34:34.395368700 +0000
@@ -169,7 +169,6 @@ PKG_CHECK_MODULES(LIBRSVG,
 	harfbuzz >= $HARFBUZZ_REQUIRED         \
 	libxml-2.0 >= $LIBXML_REQUIRED         \
 	pangocairo >= $PANGO_REQUIRED          \
-	pangoft2 >= $PANGO_REQUIRED	       \
 )
 
 CAIRO_VERSION=`$PKG_CONFIG --modversion cairo`
diff -aurp librsvg-2.48.8-orig/Makefile.am librsvg-2.48.8/Makefile.am
--- librsvg-2.50.5-orig/Makefile.am 2021-05-02 22:54:01.000000000 +0200
+++ librsvg-2.50.5/Makefile.am  2021-05-02 22:54:17.000000000 +0200
@@ -164,7 +164,7 @@ CARGO_TARGET_DIR=$(LIBRSVG_BUILD_DIR)/ta
 if LIB_NAME_WORKAROUND
 # Workaround for https://github.com/rust-lang/rust/issues/69904
 # can be removed when our minimum Rust version is at least 1.44.
-RUST_LIB=$(CARGO_TARGET_DIR)/$(RUST_TARGET_SUBDIR)/rsvg_c_api.lib
+RUST_LIB=$(CARGO_TARGET_DIR)/$(RUST_TARGET_SUBDIR)/librsvg_c_api.a
 else
 RUST_LIB=$(CARGO_TARGET_DIR)/$(RUST_TARGET_SUBDIR)/librsvg_c_api.a
 endif
diff -aurp librsvg-2.50.5-orig/Makefile.in librsvg-2.50.5/Makefile.in
--- librsvg-2.50.5-orig/Makefile.in 2021-05-02 22:54:01.000000000 +0200
+++ librsvg-2.50.5/Makefile.in  2021-05-02 22:54:22.000000000 +0200
@@ -704,7 +704,7 @@ CARGO_TARGET_DIR = $(LIBRSVG_BUILD_DIR)/

 # Workaround for https://github.com/rust-lang/rust/issues/69904
 # can be removed when our minimum Rust version is at least 1.44.
-@LIB_NAME_WORKAROUND_TRUE@RUST_LIB = $(CARGO_TARGET_DIR)/$(RUST_TARGET_SUBDIR)/rsvg_c_api.lib
+@LIB_NAME_WORKAROUND_TRUE@RUST_LIB = $(CARGO_TARGET_DIR)/$(RUST_TARGET_SUBDIR)/librsvg_c_api.a
 librsvg_@RSVG_API_MAJOR_VERSION@_la_CPPFLAGS = \
  -I$(top_srcdir)       \
  -I$(top_builddir)       \
