--- a/include/ma_global.h
+++ b/include/ma_global.h
@@ -745,7 +745,7 @@ typedef char		bool;	/* Ordinary boolean values 0 1 */
 /* Optimized store functions for Intel x86 */
 #define int1store(T,A) *((int8*) (T)) = (A)
 #define uint1korr(A)   (*(((uint8*)(A))))
-#if defined(__i386__) || defined(_WIN32)
+#if defined(__i386__) || defined(_WIN32) && !defined(__MINGW64__)
 #define sint2korr(A)	(*((int16 *) (A)))
 #define sint3korr(A)	((int32) ((((uchar) (A)[2]) & 128) ? \
 				  (((uint32) 255L << 24) | \

