diff -aurp mariadb-connector-c-2.3.7-src-orig/libmariadb/CMakeLists.txt mariadb-connector-c-2.3.7-src/libmariadb/CMakeLists.txt
--- mariadb-connector-c-2.3.7-src-orig/libmariadb/CMakeLists.txt	2018-12-30 12:24:18.322559000 +0000
+++ mariadb-connector-c-2.3.7-src/libmariadb/CMakeLists.txt	2018-12-30 12:30:50.565850600 +0000
@@ -398,13 +398,6 @@ ENDIF()
 ADD_LIBRARY(mariadbclient STATIC ${mariadbclient_RC} $<TARGET_OBJECTS:mariadb_obj> ${EMPTY_FILE} ${EXPORT_LINK})
 TARGET_LINK_LIBRARIES(mariadbclient ${SYSTEM_LIBS})
 
-ADD_LIBRARY(mariadb SHARED ${libmariadb_RC} $<TARGET_OBJECTS:mariadb_obj> ${EMPTY_FILE})
-TARGET_LINK_LIBRARIES(mariadb ${SYSTEM_LIBS})
-IF(UNIX)
-  SET_TARGET_PROPERTIES(mariadb PROPERTIES COMPILE_FLAGS "${CMAKE_SHARED_LIBRARY_C_FLAGS}")
-ENDIF()
-SIGN_TARGET(mariadb)
-
 IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
   TARGET_LINK_LIBRARIES (libmariadb "-Wl,--no-undefined")
   SET_TARGET_PROPERTIES(libmariadb PROPERTIES LINK_FLAGS "-Wl,--version-script=${EXPORT_FILE}")
@@ -412,12 +405,6 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES LINK_FLAGS "-Wl,--version-script=${EXPORT_FILE}")
 ENDIF()
 
-SET_TARGET_PROPERTIES(mariadb PROPERTIES PREFIX "")
-
-SET_TARGET_PROPERTIES(mariadb PROPERTIES VERSION 
-   ${CPACK_PACKAGE_VERSION_MAJOR}
-   SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
-
 IF(NOT WIN32)
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES OUTPUT_NAME "${LIBMARIADB_STATIC_NAME}")
 ENDIF()
@@ -442,7 +429,7 @@ ENDIF()
 
 
 INSTALL(TARGETS
-          mariadb mariadbclient
+          mariadbclient
           RUNTIME DESTINATION "${BIN_INSTALL_DIR}"
           LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
           ARCHIVE DESTINATION "${LIB_INSTALL_DIR}")
