--- mariadb-connector-c-3.1.11-src/include/ma_global.h.orig	2020-10-30 10:09:04.000000000 +0000
+++ mariadb-connector-c-3.1.11-src/include/ma_global.h	2020-10-30 10:09:04.000000000 +0000
@@ -437,7 +437,7 @@
 #define NO_PISAM		/* Not needed anymore */
 #define NO_MISAM		/* Not needed anymore */
 #define NO_HASH			/* Not needed anymore */
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
 #define NO_DIR_LIBRARY		/* Not standard dir-library */
 #define USE_MY_STAT_STRUCT	/* For my_lib */
 #ifdef _MSC_VER
