diff -aurp apache-arrow-1.0.0/cpp/CMakeLists.txt apache-arrow-1.0.0-new/cpp/CMakeLists.txt
--- apache-arrow-1.0.0/cpp/CMakeLists.txt	2020-07-20 21:14:01.000000000 +0000
+++ apache-arrow-1.0.0-new/cpp/CMakeLists.txt	2020-08-01 22:48:17.946045100 +0000
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.9)
 message(STATUS "Building using CMake version: ${CMAKE_VERSION}")
 
 # Compiler id for Apple Clang is now AppleClang.
@@ -66,6 +66,8 @@ string(TOLOWER ${CMAKE_BUILD_TYPE} LOWER
 string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_BUILD_TYPE)
 
 project(arrow VERSION "${ARROW_BASE_VERSION}")
+include(CheckIPOSupported)
+check_ipo_supported()
 
 set(ARROW_VERSION_MAJOR "${arrow_VERSION_MAJOR}")
 set(ARROW_VERSION_MINOR "${arrow_VERSION_MINOR}")
