diff -aurp apache-arrow-1.0.0-orig/cpp/cmake_modules/ThirdpartyToolchain.cmake apache-arrow-1.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake
--- apache-arrow-1.0.0-orig/cpp/cmake_modules/ThirdpartyToolchain.cmake	2020-07-20 07:14:38.327028400 +0000
+++ apache-arrow-1.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake	2020-07-20 07:17:27.392025900 +0000
@@ -1501,6 +1501,7 @@ if(ARROW_MIMALLOC)
   externalproject_add(mimalloc_ep
                       URL ${MIMALLOC_SOURCE_URL}
                       CMAKE_ARGS ${MIMALLOC_CMAKE_ARGS}
+                      PATCH_COMMAND sed -i.bak "s/601/999/g" src/os.c
                       BUILD_BYPRODUCTS "${MIMALLOC_STATIC_LIB}")
 
   include_directories(SYSTEM ${MIMALLOC_INCLUDE_DIR})
