diff -aurp v8-3.15.11.18-orig/src/liveedit.cc v8-3.15.11.18/src/liveedit.cc
--- v8-3.15.11.18-orig/src/liveedit.cc  2018-06-20 13:44:37.154720000 +0000
+++ v8-3.15.11.18/src/liveedit.cc       2018-06-20 13:52:52.134069800 +0000
@@ -203,7 +203,7 @@ class Differencer {

   static const int kDirectionSizeBits = 2;
   static const int kDirectionMask = (1 << kDirectionSizeBits) - 1;
-  static const int kEmptyCellValue = -1 << kDirectionSizeBits;
+  static const int kEmptyCellValue = -(1 << kDirectionSizeBits);

   // This method only holds static assert statement (unfortunately you cannot
   // place one in class scope).
diff -aurp v8-3.15.11.18-orig/src/objects.h v8-3.15.11.18/src/objects.h
--- v8-3.15.11.18-orig/src/objects.h	2018-06-20 13:44:36.967221900 +0000
+++ v8-3.15.11.18/src/objects.h	2018-06-20 13:48:10.792929500 +0000
@@ -5222,7 +5222,7 @@ class Map: public HeapObject {
   static const int kElementsKindBitCount = 5;
 
   // Derived values from bit field 2
-  static const int kElementsKindMask = (-1 << kElementsKindShift) &
+  static const int kElementsKindMask = -(1 << kElementsKindShift) &
       ((1 << (kElementsKindShift + kElementsKindBitCount)) - 1);
   static const int8_t kMaximumBitField2FastElementValue = static_cast<int8_t>(
       (FAST_ELEMENTS + 1) << Map::kElementsKindShift) - 1;
@@ -7413,7 +7413,7 @@ class String: public HeapObject {
   STATIC_CHECK(IS_POWER_OF_TWO(kMaxCachedArrayIndexLength + 1));
 
   static const int kContainsCachedArrayIndexMask =
-      (~kMaxCachedArrayIndexLength << kArrayIndexHashLengthShift) |
+      -(-~kMaxCachedArrayIndexLength << kArrayIndexHashLengthShift) |
       kIsNotArrayIndexMask;
 
   // Value of empty hash field indicating that the hash is not computed.
