diff -up v8-3.14.5.10/src/frames.h.gcc8 v8-3.14.5.10/src/frames.h
--- v8-3.14.5.10/src/frames.h.gcc8	2018-05-18 14:41:15.119069125 -0400
+++ v8-3.14.5.10/src/frames.h	2018-05-18 14:42:07.421810155 -0400
@@ -67,7 +67,7 @@ class InnerPointerToCodeCache {
   Code* GcSafeCastToCode(HeapObject* object, Address inner_pointer);
 
   void Flush() {
-    memset(&cache_[0], 0, sizeof(cache_));
+    memset(static_cast<void*>(&cache_[0]), 0, sizeof(cache_));
   }
 
   InnerPointerToCodeCacheEntry* GetCacheEntry(Address inner_pointer);
